Add copy to clipboard for branch/repository name

tsvetilian-ty opened the pull request:
<!--
What GitHub Desktop issue does this PR address? (for example, #1234)
-->

Closes #12141

### Screenshots

<!--
If this PR touches the UI layer of the app, please include screenshots or animated gifs to show the changes.
-->
![image](https://user-images.githubusercontent.com/9341546/117554752-47092f80-b062-11eb-97cc-462215f44153.png)
![image](https://user-images.githubusercontent.com/9341546/117554767-73bd4700-b062-11eb-84dd-9f70b929dfe2.png)

tsvetilian-ty committed:
Add copy to clipboard for branch/repository name

tsvetilian-ty committed:
Apply formatter

tidy-dev reviewed (changes requested):
@tsvetilian-ty  Thank you so much for contributing! I particularly would like this feature. 😀 

However, we have some minor changes we would like for app congruency. 

For the branch menu:

```
Rename...
Copy branch name
-----
Delete...
```

For repo menu:
```
Create alias
Copy repo name
----
Open in
Show in
Open in
----
Remove...
```

tsvetilian-ty committed:
Refactor copy to clipboard

- Update the item names to better represent the functionality
- Update items positions

tsvetilian-ty commented:
@tidy-dev the pleasure to contribute to the project is all mine 🙌 
![image](https://user-images.githubusercontent.com/9341546/117707234-fcf58a80-b1d6-11eb-8648-c188ccb6f7b5.png)
![image](https://user-images.githubusercontent.com/9341546/117707247-041c9880-b1d7-11eb-8aa2-46cd1c9af4a1.png)

tidy-dev reviewed (approved):
✨ @tsvetilian-ty Thank you so much for those changes!

clary54 commented on app/src/ui/branches/branch-list-item.tsx:
```diff
@@ -71,6 +72,13 @@ export class BranchListItem extends React.Component<IBranchListItemProps, {}> {
     }
 
+    items.push({
+      label: __DARWIN__ ? 'Copy Branch Name' : 'Copy branch name',
```
clary54
